op-e2e/system_test.go

```diff
@@ -1496,24 +1498,49 @@ func TestChallenge(t *testing.T) {
 	}
 	cancel()
 
-	// set a timeout for security council to validate output
-	ctx, cancel = context.WithTimeout(context.Background(), 10*time.Second)
+	findTransactionId := func() *big.Int {
+		validationRequestedEvent := securityCouncilABI.Events["ValidationRequested"]
+		eventIDTopic := []common.Hash{validationRequestedEvent.ID}
+		toBlock := latestBlock(t, l1Client)
+
+		query := ethereum.FilterQuery{
+			ToBlock:   big.NewInt(int64(toBlock)),
+			Addresses: []common.Address{cfg.L1Deployments.SecurityCouncilProxy},
+			Topics:    [][]common.Hash{eventIDTopic},
+		}
+
+		cCtx, cCancel := context.WithTimeout(context.Background(), 5*time.Second)
+		defer cCancel()
+		logs, err := l1Client.FilterLogs(cCtx, query)
+		require.NoError(t, err)
+		require.Equal(t, len(logs), 1)
+
+		return new(big.Int).SetBytes(logs[0].Topics[1][:])
+	}
+
+	// set a timeout for security council to validate output (provingTimeout + buffer)
+	provingTimeout, err := colosseum.PROVINGTIMEOUT(&bind.CallOpts{})
+	ctx, cancel = context.WithTimeout(context.Background(), time.Duration(provingTimeout.Uint64()+5)*time.Second)
 	defer cancel()
 
+	numCheck := 0
 	for ; ; <-ticker.C {
```

Comment:
even though the for loop is the last piece of code. This avoids referencing variables outside of the for loop.
```go
for numCheck := 0; ; <-ticker.C {
```